doc/tutorials/validation.md

```diff
@@ -133,41 +133,110 @@ kdb setmeta spec/tutorial/spec/test check/validation/match LINE
-check/validation/message = Not a number
-HERE
 kdb lsmeta /tutorial/spec/test
 #> check/validation
 #> check/validation/match
 #> check/validation/message
 ```
+Note that this key should not have passed the validation that we defined in the
+spec namespace.
+Nonetheless we were able to set this key, because the validation plugin was not
+active for this key.
+On that behalf we have to make sure that the validation plugin is loaded for
+this key with:
+```sh
+kdb mount tutorial.dump user/tutorial dump validation
+```
+This [mounts](/doc/tutorials/mount.md) the backend `tutorial.dump` to the mountpoint
+**user/tutorial** and activates the validation plugin for the keys below the mountpoint.
+The validation plugin now uses the metadata of the keys below **user/tutorial**
+to validate values before storing them in `tutorial.dump`.
+
+Although this is better than defining metadata in the same place as the data
+itself, we can still do better.
+The reason for that is that one of the aims of Elektra is to remove the trouble
+of validation and finding the files that hold your configuration from the users.
+At the moment a user still has to know which files should hold the configuration
+and which plugins must be loaded when he mounts configuration files.
+
+This problem can be addressed by recognizing that the location of the
+configuration files and the plugins that must be loaded is part of the
+_schema_ of our configuration and therefore should be stored in the spec namespace.
+
+### Specfiles ###
+We call the files, that contain a complete schema for configuration
+below a specific path in form of metadata, _Specfiles_.
+
+Particularly a _Specfile_ contains metadata that defines
+- the mountpoints of paths,
+- the plugins to load and
+- the behavior of these plugins.
+
+Let us create an example _Specfile_ in the [ni format](/doc/src/plugins/ni/README.md),
+which supports metadata:
+```sh
+kdb mount tutorial.ini spec/tutorial ni
+cat << HERE | kdb import spec/tutorial ni
+[]
+ mountpoint = tutorial.dump
+ infos/plugins = dump validation
+
+[/links/_]
+check/validation = https?://.*\..*
+check/validation/match = LINE
+check/validation/message = not a valid URL
+description = A link to some website
+HERE
+kdb lsmeta spec/tutorial
+#> infos/plugins
+#> mountpoint
+```
+We now have all the metadata that we need to mount and validate the data below
+`/tutorial` in one file.
 
-As next step we need to correctly mount all validation plugins to `/tutorial/spec`.
-Because we have an extra specification of the validation, we can use this
-specification to assemble the plugins. We only have to specify where the
-mountpoint is (and which file name should be used), then we can mount
-the file according the specification and have validation for all namespaces:
+Now we apply this _Specfile_ to the key database to all keys below `tutorial`.
+```sh
+kdb spec-mount /tutorial
+```
+This command automatically mounts `/tutorial` to the backend `tutorial.dump` and
```

Comment:
Same sentence as above?

Again: `tutorial.dump` is a file and not a backend?
Do you mean something like:

    mounts `/tutorial` as specified:

and then explain what was specified above (what the individual lines mean)